Replicator/Replicator.cc

```diff
@@ -147,6 +147,95 @@ namespace litecore { namespace repl {
         _disconnect(websocket::kCodeNormal, {});
     }
 
+    alloc_slice Replicator::pendingDocumentIDs(C4Error* outErr) const {
+        if(!_pusher) {
+            // Couchbase Lite should not allow this case
+            return nullslice;
+        }
+
+        return _db->use<alloc_slice>([this, outErr](C4Database* db)
+        {
+            const auto dbLastSequence = c4db_getLastSequence(db);
+            const auto replLastSequence = _checkpoint.sequences().local;
+            if(replLastSequence >= dbLastSequence) {
+                return alloc_slice(nullslice);
```

Comment:
Compiler nuance.  At least on Windows, it won't let you return a different type than the one specified in the template arguments.  If I change the closure to something else then it will complain about the return type of the closure itself.  This seemed like the neatest way.